hazelcast/src/main/java/com/hazelcast/cache/impl/HazelcastServerCachingProvider.java

```diff
@@ -56,83 +61,100 @@ public static HazelcastServerCachingProvider createCachingProvider(HazelcastInst
+                                                             URI uri, ClassLoader classLoader,
+                                                             Properties properties) {
         return new HazelcastServerCacheManager(this, instance, uri, classLoader, properties);
     }
 
-    private HazelcastInstance getOrCreateInstance(ClassLoader classLoader, Properties properties, boolean isDefaultURI)
+    @Override
+    protected HazelcastInstance getOrCreateInstance(URI uri, ClassLoader classLoader, Properties properties)
             throws URISyntaxException, IOException {
-        HazelcastInstance instanceItself =
-                (HazelcastInstance) properties.get(HazelcastCachingProvider.HAZELCAST_INSTANCE_ITSELF);
+        HazelcastInstance instanceItself = (HazelcastInstance) properties.get(HAZELCAST_INSTANCE_ITSELF);
 
         // If instance itself is specified via properties, get instance through it.
         if (instanceItself != null) {
             return instanceItself;
         }
 
-        String location = properties.getProperty(HazelcastCachingProvider.HAZELCAST_CONFIG_LOCATION);
-        String instanceName = properties.getProperty(HazelcastCachingProvider.HAZELCAST_INSTANCE_NAME);
+        String location = properties.getProperty(HAZELCAST_CONFIG_LOCATION);
+        String instanceName = properties.getProperty(HAZELCAST_INSTANCE_NAME);
 
         // If config location is specified via properties, get instance through it.
         if (location != null) {
-            URI uri = new URI(location);
-            String scheme = uri.getScheme();
-            if (scheme == null) {
-                // It is a place holder
-                uri = new URI(System.getProperty(uri.getRawSchemeSpecificPart()));
-            }
-            ClassLoader theClassLoader = classLoader == null ? getDefaultClassLoader() : classLoader;
-            final URL configURL;
-            if ("classpath".equals(scheme)) {
-                configURL = theClassLoader.getResource(uri.getRawSchemeSpecificPart());
-            } else if ("file".equals(scheme) || "http".equals(scheme) || "https".equals(scheme)) {
-                configURL = uri.toURL();
-            } else {
-                throw new URISyntaxException(location, "Unsupported protocol in configuration location URL");
-            }
-            try {
-                Config config = getConfig(configURL, theClassLoader, instanceName);
-                return HazelcastInstanceFactory.getOrCreateHazelcastInstance(config);
-            } catch (Exception e) {
-                throw ExceptionUtil.rethrow(e);
-            }
+            Config config = getConfigFromLocation(location, classLoader, instanceName);
+            return HazelcastInstanceFactory.getOrCreateHazelcastInstance(config);
         }
 
         // If instance name is specified via properties, get instance through it.
         if (instanceName != null) {
             return Hazelcast.getHazelcastInstanceByName(instanceName);
```

Comment:
out of the scope for this PR, but I believe this should use `getOrCreate()`

right now there is an inconsistency: when you explicitly pass a configuration then it will always use it and it will use `getOrCreate()`

I believe when you pass just the `instanceName`, but no explicit configuration then it should use default configuration.